api/handler_get_package.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"net/http"
+
+	"github.com/gophr-pm/gophr/lib/db/query"
+	"github.com/gophr-pm/gophr/lib/errors"
+	"github.com/gophr-pm/gophr/lib/model/package"
+	"github.com/gorilla/mux"
+)
+
+type getPackageRequestArgs struct {
+	repo   string
+	author string
+}
+
+// GetPackageHandler creates an HTTP request handler that responds to individual
+// package get requests.
+func GetPackageHandler(
+	q query.Queryable,
+) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var (
+			err    error
+			args   getPackageRequestArgs
+			json   []byte
+			result pkg.Details
+		)
+
+		// Parse out the args.
+		if args, err = extractGetPackageRequestArgs(r); err != nil {
+			errors.RespondWithError(w, err)
+			return
+		}
+		// Get from the database.
+		if result, err = pkg.Get(q, args.author, args.repo); err != nil {
+			errors.RespondWithError(w, err)
+			return
+		}
+		// Turn the result into JSON.
+		if json, err = result.ToJSON(); err != nil {
+			errors.RespondWithError(w, err)
+			return
+		}
+
+		respondWithJSON(w, json)
+	}
+}
+
+// extractGetPackageArgs validates and extracts the necessary parameters for a
+// get package request.
+func extractGetPackageRequestArgs(
+	r *http.Request,
+) (getPackageRequestArgs, error) {
+	var (
+		vars = mux.Vars(r)
+		args getPackageRequestArgs
+	)
+
+	if args.author = vars[urlVarAuthor]; len(args.author) < 0 {
```

Comment:
what's wrong with 
args.author = vars[urlVarAuthor] 
if len(args.author < 0 ) etc

also when will a len be less than 0?